src/java_bytecode/jar_file.cpp

```diff
@@ -48,8 +76,22 @@ void jar_filet::open(const std::string &filename)
         mz_zip_reader_get_filename(&zip, i, filename_char, filename_length);
       assert(filename_length==filename_len);
       std::string file_name(filename_char);
+
+      // non-class files are loaded in any case
+      bool add_file=!has_suffix(file_name, ".class");
+      // load .class file only if they match regex
+      if(regex_match)
+        add_file|=std::regex_match(file_name, string_matcher, regex_matcher);
+      // load .class file only if it is in the match set
+      else
+        add_file|=set_matcher.count(file_name)>0;
+      if(add_file)
+      {
+        index.push_back(file_name);
+        filtered_jar[filtered_index]=i;
+        filtered_index++;
```

Comment:
Ok, I wasn't really aware this was a public interface. Apologies for missing that. The consistency still requires too much thinking to my taste, so I would rather have changed that public interface. But that's probably just me.